pkg/utils/utils.go

```diff
@@ -255,3 +256,45 @@ func StringMapKeys(m interface{}) []string {
 	sort.Strings(keys)
 	return keys
 }
+
+// Optional returns the first optional non-zero element given as variadic argument,
+// if given, or the zero element as default.
+func Optional[T any](list ...T) T {
+	var zero T
+	for _, e := range list {
+		if !reflect.DeepEqual(e, zero) {
+			return e
+		}
+	}
+	return zero
+}
+
+// OptionalDefaulted returns the first optional non-nil element given as variadic
+// argument, or the given default element. For value types a given zero
+// argument is excepted, also.
+func OptionalDefaulted[T any](def T, list ...T) T {
+	for _, e := range list {
+		if !reflect2.IsNil(e) {
+			return e
+		}
+	}
+	return def
+}
+
+// OptionalDefaultedBool checks all args for true. If no true is given
+// the given default is returned.
+func OptionalDefaultedBool(def bool, list ...bool) bool {
+	for _, e := range list {
+		if e {
+			return e
+		}
+	}
+	return def
+}
```

Comment:
I think based on the new implementation it's equivalent to `Optional(...)` if the argument list is a boolean list, because `var zero T` will be `false` and because of `!reflect.DeepEqual(e, zero)`, it will return `true` only if at least one entry is `true`.

Reply:
yes, but the default value is different.  And it is not the same behaviour like `OptionalDefaulted` because bool is not nil-able, and if should provide true, if one argument is true.

Reply:
Well, I don't really see the reason to use `OptionalDefaultedBool(true, ...)` at all. That just means `true` in all cases. So the only use case is `OptionalDefaultedBool(false, ...)` and in that case it's the same as `Optional(...)` with a bool list.

Why I don't see practical?

We have two cases:

1. `OptionalDefaultedBool(true, list...)` where `list` contains a `true` value. In this case the function will return `true`
2. `OptionalDefaultedBool(false, list...)` where `list` contains only `false` values. In this case, the function will return `true`.

So in any cases, if the first argument is `true`, it will always return `true`.

Reply:
Yes, that is correct, but if the default is determined by an expression and not set by a constant, it makes sense again.
Just have a look at the method following this one (`GetOptionFlag`).

The cases are different:
- for 1) it returns always true, regardless of the given additional arguments
- for 2) it returns true at least one true is given, otherwise it returns false.